### ISSUE TITLE
New Fatal Error with REFramework/Sunbreak v.11.0.1.0

### ISSUE THREAD

**Issue by Matadurr** [praydog/REFramework#494]
All versions of REFramework I've tested, including the most recent one, are causing a fatal error upon startup. This occurs before the splash screen JPEG can even show. This error always occurs when the dinput8.dll file is in the game's root folder, and it does not matter if there are .pak patches or other mods present, or even if it is a clean install entirely. It also doesn't matter if I am starting in DX12 or DX11. 

The crash occurs before REFramework can even initialize, so there is no REFramework crash report to upload, but please see the log file if there's any good info in there.

The error info box is also uploaded. I hope this info helps.
![Screenshot (1)](https://user-images.githubusercontent.com/111039191/184071075-2ab27138-a134-44f0-9ae8-fb876cb9b867.png)
[re2_framework_log.txt](https://github.com/praydog/REFramework/files/9306335/re2_framework_log.txt)

**Comment by jadada** [comment]
Same issue. If I remove the `dinput8.dll`, the game launch normally. I have reinstalled the game, the mod worked for the first time but failed everytime I try to launch the game with the "284 Fatal D3D error" given. Here is my reframework log:
```txt
[2022-08-11 14:25:06.160] [REFramework] [info] REFramework entry
[2022-08-11 14:25:06.162] [REFramework] [info] Game Module Addr: 140000000
[2022-08-11 14:25:06.162] [REFramework] [info] Game Module Size: 28222000
[2022-08-11 14:25:06.162] [REFramework] [info] Preallocated buffer: 13fff0fc0
[2022-08-11 14:25:06.162] [REFramework] [info] Getting OS version information...
[2022-08-11 14:25:06.162] [REFramework] [info] OS Version Information
[2022-08-11 14:25:06.162] [REFramework] [info] 	Major Version: 10
[2022-08-11 14:25:06.162] [REFramework] [info] 	Minor Version: 0
[2022-08-11 14:25:06.162] [REFramework] [info] 	Build Number: 22000
[2022-08-11 14:25:06.162] [REFramework] [info] 	Platform Id: 2
[2022-08-11 14:25:06.162] [REFramework] [info] Disclaimer: REFramework does not send this information to the developers or any other third party.
[2022-08-11 14:25:06.162] [REFramework] [info] This information is only used to help with the development of REFramework.
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\ntdll.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\kernel32.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\kernelbase.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\apphelp.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\advapi32.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\msvcrt.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\sechost.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\rpcrt4.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\ucrtbase.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\gdi32.dll
[2022-08-11 14:25:06.167] [REFramework] [info] Skipping c:\windows\system32\win32u.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Creating new node for e:\game\steam\steamapps\common\monsterhunterrise\amd_ags_x64.dll (0x7ffbba130000)
[2022-08-11 14:25:06.168] [REFramework] [info] Done e:\game\steam\steamapps\common\monsterhunterrise\amd_ags_x64.dll -> e:\game\steam\steamapps\common\monsterhunterrise\_storage_\amd_ags_x64.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\gdi32full.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\msvcp_win.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\user32.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\d3d11.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\imm32.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\ole32.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\concrt140.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\dbghelp.dll
[2022-08-11 14:25:06.168] [REFramework] [info] Skipping c:\windows\system32\combase.dll
[2022-08-11 14:25:06.172] [REFramework] [info] Creating new node for e:\game\steam\steamapps\common\monsterhunterrise\dinput8.dll (0x7ffb3dbb0000)
[2022-08-11 14:25:06.172] [REFramework] [info] Done e:\game\steam\steamapps\common\monsterhunterrise\dinput8.dll -> e:\game\steam\steamapps\common\monsterhunterrise\_storage_\dinput8.dll
[2022-08-11 14:25:06.172] [REFramework] [info] Skipping c:\windows\system32\oleaut32.dll
[2022-08-11 14:25:06.172] [REFramework] [info] Skipping c:\windows\system32\setupapi.dll
[2022-08-11 14:25:06.172] [REFramework] [info] Skipping c:\windows\system32\dxgi.dll
[2022-08-11 14:25:06.172] [REFramework] [info] Skipping c:\windows\system32\shell32.dll
[2022-08-11 14:25:06.172] [REFramework] [info] Skipping c:\windows\system32\shlwapi.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\comdlg32.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\ws2_32.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\shcore.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\hid.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\iphlpapi.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\mfplat.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\mfreadwrite.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\vcruntime140.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\msvcp140.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\vcruntime140_1.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\winhttp.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\version.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\winmm.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\wininet.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\winsxs\amd64_microsoft.windows.common-controls_6595b64144ccf1df_5.82.22000.1_none_271a8fad6a2d1b1e\comctl32.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\cfgmgr32.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\d3dcompiler_47.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\cryptsp.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\xinput1_3.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\kernel.appcore.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\bcryptprimitives.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\rtworkq.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping e:\game\steam\gameoverlayrenderer64.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\psapi.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\dinput8.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\inputhost.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\coremessaging.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\d3d12.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\cryptbase.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping c:\windows\system32\ntmarta.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping e:\game\steam\steamapps\common\monsterhunterrise\_storage_\amd_ags_x64.dll
[2022-08-11 14:25:06.173] [REFramework] [info] Skipping e:\game\steam\steamapps\common\monsterhunterrise\_storage_\dinput8.dll
[2022-08-11 14:25:06.173] [REFramework] [info] [PluginLoader] Module path E:\Game\Steam\steamapps\common\MonsterHunterRise\MonsterHunterRise.exe
[2022-08-11 14:25:06.173] [REFramework] [info] [PluginLoader] Creating directories E:\Game\Steam\steamapps\common\MonsterHunterRise\reframework\plugins
[2022-08-11 14:25:06.174] [REFramework] [info] [PluginLoader] Loading plugins...
[2022-08-11 14:25:11.241] [REFramework] [info] Last chance encountered for hooking
[2022-08-11 14:25:12.248] [REFramework] [info] Sending rehook request for D3D
[2022-08-11 14:25:12.251] [REFramework] [info] Hooking D3D12
[2022-08-11 14:25:12.252] [REFramework] [info] Creating dummy device
[2022-08-11 14:25:12.255] [REFramework] [info] D3D12CreateDevice appears to be hooked, temporarily unhooking
[2022-08-11 14:25:15.993] [REFramework] [info] Restoring hooked bytes for D3D12CreateDevice
[2022-08-11 14:25:15.993] [REFramework] [info] Dummy device: 5e171da0
[2022-08-11 14:25:15.995] [REFramework] [info] Creating dummy DXGI factory
[2022-08-11 14:25:15.996] [REFramework] [info] Creating dummy command queue
[2022-08-11 14:25:16.013] [REFramework] [info] Creating dummy swapchain
[2022-08-11 14:25:16.013] [REFramework] [info] Trying swapchain attempt 0
[2022-08-11 14:25:16.065] [REFramework] [info] Created dummy swapchain on attempt 0
[2022-08-11 14:25:16.065] [REFramework] [info] Querying dummy swapchain
[2022-08-11 14:25:16.065] [REFramework] [info] Finding command queue offset
[2022-08-11 14:25:16.065] [REFramework] [info] Found command queue offset: 118
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 13164
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 16716
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 3528
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 13516
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 11428
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 3608
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 2364
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 2444
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 5592
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 1780
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 15572
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 4076
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 4312
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 2252
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 16780
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 14620
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 16108
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 15624
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 4700
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 9348
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 688
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 8632
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 14884
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 13356
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 5744
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 8220
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 12060
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 17276
[2022-08-11 14:25:16.083] [REFramework] [info] Suspending 10536
[2022-08-11 14:25:16.084] [REFramework] [info] Suspending 15456
[2022-08-11 14:25:16.084] [REFramework] [info] Suspending 2760
[2022-08-11 14:25:16.084] [REFramework] [info] Initializing hooks
[2022-08-11 14:25:16.084] [REFramework] [info] [PointerHook] Hooking 7ffbbd7fde18->7ffbbd7614c0 to 7ffb3dbb9640
[2022-08-11 14:25:16.084] [REFramework] [info] [PointerHook] Hooking 7ffbbd7fde40->7ffbbd787790 to 7ffb3dbb9880
[2022-08-11 14:25:16.084] [REFramework] [info] [PointerHook] Hooking 7ffbbd7fde48->7ffbbd7d6b10 to 7ffb3dbb9b20
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 13164
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 16716
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 3528
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 13516
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 11428
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 3608
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 2364
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 2444
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 5592
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 1780
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 15572
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 4076
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 4312
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 2252
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 16780
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 14620
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 16108
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 15624
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 4700
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 9348
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 688
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 8632
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 14884
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 13356
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 5744
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 8220
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 12060
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 17276
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 10536
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 15456
[2022-08-11 14:25:16.084] [REFramework] [info] Resuming 2760
[2022-08-11 14:25:16.095] [REFramework] [info] Hooked DirectX 12
[2022-08-11 14:25:18.658] [REFramework] [error] Failed to initialize REFramework on DirectX 12
[2022-08-11 14:25:18.659] [REFramework] [info] Present fixed
[2022-08-11 14:25:18.664] [REFramework] [info] D3D12 resize target called
[2022-08-11 14:25:18.664] [REFramework] [info] Reset!
[2022-08-11 14:25:18.917] [REFramework] [info] D3D12 resize target called
[2022-08-11 14:25:18.917] [REFramework] [info] Reset!
[2022-08-11 14:25:19.242] [REFramework] [info] D3D12 resize buffers called
[2022-08-11 14:25:19.242] [REFramework] [info] Reset!
[2022-08-11 14:25:19.242] [REFramework] [info] D3D12 resize buffers called
[2022-08-11 14:25:19.244] [REFramework] [info] Resize buffers fixed
[2022-08-11 14:25:19.295] [REFramework] [error] Failed to initialize REFramework on DirectX 12
[2022-08-11 14:25:25.033] [REFramework] [info] REFramework shutting down...

```

**Comment by Matadurr** [comment]
> Same issue. If I remove the `dinput8.dll`, the game launch normally. I have reinstalled the game, the mod worked for the first time but failed with the "284 Fatal D3D error" given everytime I try to launch the game. So I think the issue must be related to the Reframework mod. I am using Windows 11. Here is my reframework log:

For sure, the moment I move dinput8.dll away, the game launches fine, even other .pak patch mods all work correctly. I just hope we can nail down what exactly is causing this, because this issue seems to be very rare. I feel like I’ve tried pretty much everything at this point apart from clean installing steam itself or even Windows 11.

**Comment by praydog** [comment]
I'm currently looking into it. From what I can tell, it's from `IDXGISwapChain::ResizeBuffers` returning 0x887A0001 AKA `DXGI_ERROR_INVALID_CALL`. The commonality between your crashes is that an overlay (probably the Steam overlay) is causing REFramework to detect something that would crash the game, and it attempts to fix it, but fails for some reason and causes the game to throw an error.

tl;dr I think I have an idea of what is wrong and will continue to try and come up with a solution.

**Comment by praydog** [comment]
Please try the latest build from here: https://github.com/praydog/REFramework-nightly/releases/tag/latest

Upload your log again if the issue still occurs.

**Comment by Matadurr** [comment]
> Please try the latest build from here: https://github.com/praydog/REFramework-nightly/releases/tag/latest
> 
> Upload your log again if the issue still occurs.

Please see attached. First attempt, the splash screen loaded (but only at 50% size??), and REFramework started to initialized. After splash screen, game crashed. This has been common lately, it may work on the second attempt. Github won't let me upload the crash .dmp file.
[re2_framework_log.txt](https://github.com/praydog/REFramework/files/9312777/re2_framework_log.txt)
.

**Comment by Matadurr** [comment]
Second attempt the same thing happened, so I removed the old REFramework folder that houses all the scripts/plugins I used. Still crashed after splash screen that loaded at 50% screen size.

Third attempt, the splash screen transitioned to an ultrawide ratio for some reason. Title screen loaded, and REFramework loaded. See log info if there's any decent data still in there from the second crash.

If it helps with anything, I am running my computer through an elgado 4K60S+ capture card, so that may be what's making the resolution initially screwy. After changing it manually to 4k/60p + v-sync it all looks normal.

I'm going to go through and manually add back my scripts and plugins to see what might be causing the issue. Or I may just re-download all of them again to ensure they're on the latest version.
[re2_framework_log.txt](https://github.com/praydog/REFramework/files/9312869/re2_framework_log.txt)
.

**Comment by jadada** [comment]
> Please try the latest build from here: https://github.com/praydog/REFramework-nightly/releases/tag/latest
> 
> Upload your log again if the issue still occurs.

The latest build works for me. I try to close and relaunch the game for serveral times. The game launches normally now. Thanks a lot !

**Comment by AkumaSann** [comment]
I am still crashing before reaching the start menu. No REframework menu loads either. When the dinput8 is removed game loads up fine. I've already tried new updated files. I can't get you the crash report because I don't get far enough for the mod to create its folders and what not. Fresh install only reframework installed.

**Comment by praydog** [comment]
> I am still crashing before reaching the start menu. No REframework menu loads either. When the dinput8 is removed game loads up fine. I've already tried new updated files. I can't get you the crash report because I don't get far enough for the mod to create its folders and what not. Fresh install only reframework installed.

You should have an `re2_framework_log.txt`

**Comment by AkumaSann** [comment]
I don't. Weird. Wonder what the hell is going on. I found another fix on a different thread on here that's similar to my issues. It's everything ironed out in your end? One I get home ill try and see if I have that .txt again and if I do I'll post it

**Comment by AkumaSann** [comment]
I get to the "loading do not quit the game" and it hard crashes. But I don't get the re2.txt for the crash    I have the dll installed correctly. I'm pretty lost at what's going wrong.

**Comment by AkumaSann** [comment]
Okay. All I needed was a command line overriding the default dll. Or some shit. At the moment I have it launching and running. Game a RFW UI.

**Comment by lingsamuel** [comment]
I found that this also happens when loading fonts dynamically, especially when loading fonts based on changed language config. But if I load all fonts at the same time at the script top, the problem no longer occurs.

**Comment by x3Karma** [comment]
> Okay. All I needed was a command line overriding the default dll. Or some shit. At the moment I have it launching and running. Game a RFW UI.

sorry for nercoing but could you elaborate more on how you override the default dll using the command line?

**Comment by danielselma93** [comment]
I still have the problem
[re2_framework_log.txt](https://github.com/praydog/REFramework/files/10610684/re2_framework_log.txt)
here my log i need help pls
Thanks.

**Comment by KsaNL** [comment]
The problem still exist, my friend caused this too. 
I think maybe need to check the dx12 version to prevent that
